### PR TITLE
[#1] Implement basic API

### PR DESCRIPTION
**Pull request by hangpark**
See #1

### CODE REVIEW

File: server/mapping/models.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+class PhotoQuerySet(models.QuerySet):
+    def accessible_for(self, user):
+        q = Q(is_public=True)
+        q |= Q(user__in=get_friends(user))
+        return self.filter(q).distinct()
+
+    def filter_friends(self, user):
+        return self.filter(user__in=self.get_friends(user))
```
Comment (JOYN28) [inline server/mapping/models.py]: Remove `self` in `user__in=self.get_friends(user)`.

File: server/mapping/views.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            social_auth__uid__in=id_list)
+
+        try:
+            profile = user.userprofile
+            profile.friends.clear()
+            profile.friends.add(user)
+        except:
+            profile = UserProfile(user=user)
+            profile.save()
+            profile.friends.add(user)
```
Comment (JOYN28) [inline server/mapping/views.py]: `profile.friends.add(user)` is duplicated.

File: server/mapping/views.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        targets = res.json()['data']
+        id_list = [target['id'] for target in targets]
+        friends = User.objects.filter(
+            social_auth__provider='facebook',
+            social_auth__uid__in=id_list)
+
+        try:
+            profile = user.userprofile
+            profile.friends.clear()
+            profile.friends.add(user)
```
Comment (JOYN28) [inline server/mapping/views.py]: `profile.friends.add(user)` is duplicated.

File: server/mapping/models.py (hunk 1 above)
Comment (hangpark) [inline server/mapping/models.py]: Thanks :+1:

File: server/mapping/views.py (hunk 3 above)
Comment (hangpark) [inline server/mapping/views.py]: Approved.

File: server/mapping/views.py (hunk 2 above)
Comment (hangpark) [inline server/mapping/views.py]: Approved.

**Pull request merged**